src/wallet/output_alias.c

```diff
@@ -0,0 +1,423 @@
+    cond_blk_free(state);
+    cond_blk_free(governor);
+    cond_blk_list_free(cond_blocks);
+    output_alias_free(alias_output);
+    return -1;
+  }
+
+  cond_blk_free(state);
+  cond_blk_free(governor);
+  cond_blk_list_free(cond_blocks);
+  output_alias_free(alias_output);
+
+  return 0;
+}
+
+int wallet_alias_create_send(iota_wallet_t* w, address_t* sender_addr, ed25519_keypair_t* sender_keypair,
+                             uint64_t const send_amount, address_t* state_ctrl_addr, address_t* govern_addr,
+                             address_t* alias_addr, res_send_message_t* msg_res) {
+  if (w == NULL || sender_addr == NULL || sender_keypair == NULL || state_ctrl_addr == NULL || govern_addr == NULL ||
+      alias_addr == NULL || msg_res == NULL) {
+    printf("[%s:%d] invalid parameters\n", __func__, __LINE__);
+    return -1;
+  }
+
+  int ret = 0;
+  utxo_outputs_list_t* unspent_outputs = NULL;
+  signing_data_list_t* sign_data = signing_new();
+
+  // create a tx
+  transaction_payload_t* tx = tx_payload_new(w->network_id);
+  if (!tx) {
+    printf("[%s:%d] create tx payload failed\n", __func__, __LINE__);
+    ret = -1;
+    goto end;
+  }
+
+  // get outputs from the sender address
+  uint64_t total_unspent_amount = 0;
+  unspent_outputs = wallet_get_unspent_basic_outputs(w, sender_addr, sender_keypair, send_amount, tx->essence,
+                                                     &sign_data, &total_unspent_amount);
+  if (!unspent_outputs) {
+    printf("[%s:%d] address does not have any unspent basic outputs\n", __func__, __LINE__);
+    ret = -1;
+    goto end;
+  }
+
+  // check balance of sender outputs
+  if (total_unspent_amount < send_amount) {
+    printf("[%s:%d] insufficient address balance\n", __func__, __LINE__);
+    ret = -1;
+    goto end;
+  }
+
+  // create alias output
+  byte_t alias_id[ALIAS_ID_BYTES] = {0};
+  ret = wallet_output_alias_create(tx->essence, alias_id, 0, state_ctrl_addr, govern_addr, send_amount);
+  if (ret != 0) {
+    printf("[%s:%d] create alias output failed\n", __func__, __LINE__);
+    goto end;
+  }
+
+  // check if reminder is needed
+  if (total_unspent_amount > send_amount) {
+    ret = wallet_output_basic_create(sender_addr, total_unspent_amount - send_amount, tx->essence);
+    if (ret != 0) {
+      printf("[%s:%d] create a reminder basic output failed\n", __func__, __LINE__);
+      goto end;
+    }
+  }
+
+  // create a core message
+  core_message_t* message = wallet_create_core_message(w, tx, unspent_outputs, sign_data);
+  if (!message) {
+    printf("[%s:%d] can not create a core message\n", __func__, __LINE__);
+    goto end;
+  }
+
+  // calculate transaction payload ID
+  byte_t payload_id[CRYPTO_BLAKE2B_256_HASH_BYTES] = {0};
+  if (tx_payload_calculate_id(tx, payload_id, sizeof(payload_id)) != 0) {
+    ret = -1;
+    printf("[%s:%d] can not calculate transaction payload ID\n", __func__, __LINE__);
+    goto end;
```

Comment:
message is not freed at the end.
```suggestion
    core_message_free(message);
    printf("[%s:%d] can not calculate transaction payload ID\n", __func__, __LINE__);
    goto end;
```